python/pyarrow/tests/test_dataset.py

```diff
@@ -981,6 +981,48 @@ def test_make_fragment(multisourcefs):
         assert row_group_fragment.row_groups == [0]
 
 
+@pytest.mark.parquet
+@pytest.mark.s3
+def test_make_fragment_with_size(s3_example_simple):
```

Comment:
We don't need S3 for these tests, do we? Can we simply use the local filesystem, or even a mock filesystem?

Reply:
Good point, I was focused on an S3 use case so didn't consider this. I tried to convert this to use the same mock filesystem from `multisourcefs` as the above test, but in that case the exceptions do not get raised, and the `to_table()` calls return incomplete data of 24 rows (in both size=1 and size=1000000). Need to look into this more to see if it's a bug on the Python side or the C++ side.

Reply:
Looked into this further, the incomplete data issue was just a bug in the test so a false alarm. But basically this test can't use local or mock filesystems because they don't currently implement `OpenInputFile` that takes `FileInfo`, so the size argument just gets ignored. Given this, do you think it would be OK to just use S3 here, as that is one concrete implementation in which this optimization is used?

Reply:
@eeroel Sorry for the delay. Yes, this sounds ok to me. Can you add a comment explaining this?